From 0be7fa32b54f42700955a9499b7632f5d365920a Mon Sep 17 00:00:00 2001
From: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Date: Thu, 12 Apr 2018 00:16:18 +0700
Subject: [PATCH 478/504] usb: host: xhci-plat: treat r8a7744 with the same
 xhci quirk as r8a7743

R8A7744 (RZ/G1N) platform has the same xhci quirk to R8A7743 (RZ/G1M)
and cannot work with 4GB phy devices. Therefore, we should apply similar
treatment in the commit (usb: host: xhci-plat: fix cannot work if R-Car
Gen2/3 run on above 4GB phys) to R8A7744.

Signed-off-by: thongsyho <thong.ho.px@rvc.renesas.com>
Signed-off-by: Kha Tran <kha.tran.px@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynhnguyen.xb@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 drivers/usb/host/xhci-plat.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index 4305bcc..4dc008b 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -54,7 +54,8 @@ static void xhci_plat_quirks(struct device *dev, struct xhci_hcd *xhci)
 	 * to call dma_set_coherent_mask(dev, DMA_BIT_MASK(32)) in
 	 * xhci_gen_setup().
 	 */
-	if (of_device_is_compatible(of_node, "renesas,xhci-r8a7743"))
+	if (of_device_is_compatible(of_node, "renesas,xhci-r8a7743") ||
+	    of_device_is_compatible(of_node, "renesas,xhci-r8a7744"))
 		xhci->quirks |= XHCI_NO_64BIT_SUPPORT;
 }
 
-- 
2.7.4

